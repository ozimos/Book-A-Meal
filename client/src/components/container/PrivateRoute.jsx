import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import validateToken from '../../services/validateToken';

const user = JSON.parse(localStorage.getItem('user'));
let result = {};
if (user && (user.constructor === Object) && (Object.keys(user).length !== 0)) {
  result = validateToken(user);
}

const PrivateRoute = ({ caterer: Caterer, customer: Customer, ...rest }) => (
  <Route
    {...rest}
    render={(props) => {
      if (result.valid && user.data.isCaterer) {
        return (<Caterer {...props} />);

      } else if (result.valid && !user.data.isCaterer) {
        return (<Customer {...props} />);
      } return (<Redirect to="/login" />);
    }}
  />
);
PrivateRoute.defaultProps = {
  caterer: null,
  customer: null,
};
PrivateRoute.propTypes = {
  caterer: PropTypes.func,
  customer: PropTypes.func,
};
export default PrivateRoute;
