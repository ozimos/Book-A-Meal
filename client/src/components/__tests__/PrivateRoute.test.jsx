/* global React:false, shallow:false toJson:false */

import PrivateRoute from '../container/PrivateRoute';


describe('PrivateRoute Component', () => {
  jest.mock('react-router-dom', () => (
    {
      Route: 'div',
      Redirect: 'div',

    }
  ));

  it('renders correctly', () => {
    const wrapper = shallow(<PrivateRoute
      component={jest.fn()}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

