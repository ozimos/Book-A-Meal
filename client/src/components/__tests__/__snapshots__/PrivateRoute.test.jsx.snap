// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute Component renders correctly 1`] = `
<Route
  component={[MockFunction]}
  render={[Function]}
/>
`;
